### DISCUSSION TITLE
Error: Request failed with status code 429

### DISCUSSION THREAD

**Discussion by rfash** [AntlerVC/firetable#1261] [Support / Q&A] [answered]
I get a 429 error when using ParallelGPT to process about 20,0000 records imported with CSV.
About 6000 of them processed as expected and the remaining 14000 show nothing in the chatGptResponse.

When i look at the logs I see the following; 

jsonPayload:{6 items
fieldName:"chatGptResponse"
functionType:"derivative-function"
loggingSource:"backend-function"
payload:{1 item
error:"Error: Request failed with status code 429"
}
rowId:"xxxxx"
tablePath:"xxxxx/xxxxxx"
}

Also, are there limits on how many records should be processed at a go ?

**Comment by BeeBombshell** [comment, marked as answer]
Hey @rfash, thanks for reaching out. The 429 error you encountered indicates that you have exceeded the rate limit for the GPT-3 API. This error occurs when you make too many requests within a certain time frame. The GPT-3 API has rate limits in place to ensure fair usage and maintain system stability.

If you need to process a large number of records, consider batching them into smaller groups and processing each group separately. This allows you to spread the requests over time and avoid hitting the rate limits.
